dev/NavigationView/NavigationView.cpp

```diff
@@ -4700,6 +4735,65 @@ bool NavigationView::IsFullScreenOrTabletMode()
     return isFullScreenMode || isTabletMode;
 }
 
+void NavigationView::SetDropShadow()
+{
+    const auto shadowCaster = m_shadowCaster.get();
+    const auto displayMode = DisplayMode();
+
+    if (displayMode == winrt::NavigationViewDisplayMode::Compact || displayMode == winrt::NavigationViewDisplayMode::Minimal)
+    {
+        if (shadowCaster)
+        {
+            const auto rootSplitView = m_rootSplitView.get();
+            const auto rootSplitViewActualWidth = rootSplitView.ActualWidth();
+
+            // If the screen real estate is too small, the left pane of SplitView ends up being smaller than OpenPaneLength and takes the width of RootSplitView.
+            // This requires a different storyboard animation than default.
+            if (rootSplitViewActualWidth < rootSplitView.OpenPaneLength())
+            {
+                // Since the width of the NavigationView pane is now different than OpenPaneLength, we need to update its TranslateX animation value to a reflect that.
+                const auto negativeSplitViewWidthMinusCompactLength = (rootSplitViewActualWidth - rootSplitView.CompactPaneLength()) * -1;
+                GetTemplateSettings()->NegativeSplitViewWidthMinusCompactLength(negativeSplitViewWidthMinusCompactLength);
+
+                if (const auto shadowCasterSmallPaneEaseInStoryboard = m_shadowCasterSmallPaneEaseInStoryboard.get())
+                {
+                    shadowCasterSmallPaneEaseInStoryboard.Begin();
+                }
+            }
+            else
+            {
+                if (const auto shadowCasterEaseInStoryboard = m_shadowCasterEaseInStoryboard.get())
+                {
+                    shadowCasterEaseInStoryboard.Begin();
+                }
+            }
+
+            if (winrt::IUIElement10 shadowCaster_uiElement10 = shadowCaster)
+            {
+                shadowCaster_uiElement10.Shadow(winrt::ThemeShadow{});
+            }
+        }              
+    }
+}
+
+void NavigationView::UnsetDropShadow()
+{
+    const auto shadowCaster = m_shadowCaster.get();
+
+    if (const auto shadowCasterEaseOutStoryboard = m_shadowCasterEaseOutStoryboard.get())
+    {
+        shadowCasterEaseOutStoryboard.Begin();
+    }
+
+    if (winrt::IUIElement10 shadowCaster_uiElement10 = shadowCaster)
+    {
+        if (shadowCaster_uiElement10.Shadow())
+        {
+            shadowCaster_uiElement10.Shadow(nullptr);
```

Comment:
>shadowCaster_uiElement10 [](start = 12, length = 24)

Don't we need the shadowCasterEaseOutStoryboard to complete before we remove the shadow?

Reply:
Added an async call when the storyboard has completed